src/allmydata/web/verifynode.py

```diff
@@ -0,0 +1,242 @@
+
+import simplejson
+
+from twisted.internet import defer
+from nevow import url, rend
+from nevow.inevow import IRequest
+
+from allmydata.interfaces import ExistingChildError, IFileNode
+from allmydata.monitor import Monitor
+from allmydata.mutable.publish import MutableFileHandle
+from allmydata.mutable.common import MODE_READ
+from allmydata.util import base32
+from allmydata.util.encodingutil import quote_output
+
+from allmydata.web.common import text_plain, WebError, RenderMixin, \
+     boolean_of_arg, get_arg, \
+     parse_replace_arg, parse_offset_arg, \
+     get_filenode_metadata
+from allmydata.web.check_results import CheckResultsRenderer, \
+     CheckAndRepairResultsRenderer, LiteralCheckResultsRenderer
+from allmydata.web.info import MoreInfo
+
+
+class VerifyNodeHandler(RenderMixin, rend.Page):
+    """
+    I handle requests for verifiable nodes.
+    """
+
+    def __init__(self, client, node, parentnode=None, name=None):
+        rend.Page.__init__(self)
+        self.client = client
+        assert node
+        self.node = node
+        self.parentnode = parentnode
+        self.name = name
+
+    def childFactory(self, ctx, name):
+        raise WebError("Verifycaps can't have children, certainly not named %s"
+                       % quote_output(name, encoding='utf-8'))
+
+    def render_HEAD(self, ctx):
+        # Just delegate to GET; we have to do pretty much the same work.
+        return self.render_GET(ctx)
+
+    def render_GET(self, ctx):
```

Comment:
It seems semantically wrong for ``render_{GET,PUT,POST}`` to be implemented on ``VerifyNodeHandler``, especially with a bunch of conditional logic that tries to handle all cases (mutable, read-only, etc.) for subclasses. Then again, it seems semantically wrong for these methods to have been on ``FileNodeHandler`` prior to this PR. I'm not sure what to do about this.